services/s3/src/main/java/software/amazon/awssdk/services/s3/checksums/ChecksumsEnabledValidator.java

```diff
@@ -107,6 +107,10 @@ public static boolean shouldRecordChecksum(SdkRequest sdkRequest,
             return false;
         }
 
+        if (executionAttributes.getOptionalAttribute(SdkExecutionAttribute.RESOLVED_CHECKSUM_SPECS).isPresent()) {
+            return false;
+        }
```

Comment:
Not familiar with this whole feature so can you clarify why it's false here? Is it because if the checksum specs are present then we do whatever the specs require instead of assuming it's a trailing checksum?

Reply:
S3 has existing CRC32 checksum validation for PutObject , the ChecksumsEnabledValidator is used to enable it.
Whenever HttpChecksum feature is enabled we should skip PutObject Specific Checksum validation as this will be done in the generic validators when the feature is enabled.